src/app/qgsfeatureaction.cpp

```diff
@@ -272,8 +246,6 @@ QgsFeatureAction::AddFeatureResult QgsFeatureAction::addFeature( const QgsAttrib
     if ( highlight )
       dialog->setHighlight( highlight.release() );
 
-    connect( dialog->attributeForm(), &QgsAttributeForm::featureSaved, this, &QgsFeatureAction::onFeatureSaved );
```

Comment:
Looks like there's still logic in onFeatureSaved which is used, e.g. setting mFeatureSaved.